stytra/stimulation/__init__.py

```diff
@@ -109,10 +113,19 @@ def update_protocol(self):
         self.stimuli = self.protocol._get_stimulus_list()
 
         self.current_stimulus = self.stimuli[0]
+        
+        if hasattr(self.experiment, 'estimator'):
+            self.environment_state.estimator = self.experiment.estimator
```

Comment:
If the experiment class has initialized the estimator pass it to the env_state variable

Reply:
The more "pythonic" way is try/catch an `AttributeError`.
In general but this whole logic should be isolated from the ProtocolRunner and handled by either by  `Experiemnt.get_environment_state()`. Also think about how after separating the stimulation in another process, there will be an `initialize` and `update` (from queue) methods for the `EnvironmentState`